src/actions/spring.js

```diff
@@ -0,0 +1,93 @@
+/*
+  The closed-form damped harmonic oscillating spring.
+  Or, spring.
+
+  This is a direct port of Adam Miskiewicz's (@skevy) React Animated
+  PR #15322 https://github.com/facebook/react-native/pull/15322/
+
+  ```
+  spring({
+    mass: 2,
+    damping: 10,
+    stiffness: 100,
+    to: 100
+  }).start();
+  ```
+
+  Adam Miskiewicz:
+  @skevy (twitter.com/skevy, github.com/skevy)
+ */
+import Action from './';
+import { timeSinceLastFrame } from '../framesync';
+
+class Spring extends Action {
+  static defaultProps = {
+    stiffness: 100,
+    damping: 10,
+    mass: 1.0,
+    velocity: 0.0,
+    from: 0.0,
+    to: 0.0,
+    restSpeed: 0.01,
+    restDisplacement: 0.01
+  };
+
+  onStart() {
+    const { velocity, to } = this.props;
+    this.t = 0;
+    this.initialVelocity = velocity / 1000;
```

Comment:
I've been using your example of `{ stiffness: 1000, damping: 500, mass: 3 }` as a control, as I know that's a spring that's meant to look normal.

The `velocity` calculation in this update loop does seem to output values +/-0-10 and it works fine if I feed this value back into the next spring (like React Animated).

However we use a standard velocity calculation across every action for interoperability: `speedPerSecond(current - prev, timeDelta)`

The numbers I've historically received from that are closer to +/-500-5000 unit/sec. This kind of magnitude makes more sense to me because if we're making a spring that moves from 0 - 800 and it takes ~ half a second to do so, then at it's fastest you're expecting a velocity of at least ~ 1600 px/sec rather than something in the 10s.